### PR TITLE
fix(api): return appropriate status for unauthorized requests

### LINKED ISSUE #1095: Watchtower API Mode always returns 200 OK status code
**Describe the bug**
When using watchtower in API mode sending any request to `v1/update` endpoint it will always return status `200 OK`.

**To Reproduce**
1. Setup watchtower in API mode
2. Make a request to `http://ip:port/v1/update` with missing/invalid `Authorization` header
3. Look at response code

**Expected behavior**
The API returns `401 Unauthorized` whenever the `Authorization` header is missing/invalid

**Screenshots**
image

**Environment**
Unrelated

<details>
 <summary><b> Logs from running watchtower with the <code>--debug</code> option </b></summary>

```
"Updates triggered by HTTP API request." only triggered when providing a valid token
```

</details>

**Additional context**
Would be nice to have meaningful status codes instead of just silently failing

### PR DESCRIPTION
**Pull request by hypnoglow**
Fixes #1095

Also added the test for both cases 😊

### CODE REVIEW

**Comment by simonaronsson** [comment]
Excellent work! Thank you! I'll have a more thorough look at the code as soon as I get an opportunity.

File: pkg/api/api_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+			rec := httptest.NewRecorder()
+			req := httptest.NewRequest("GET", "/hello", nil)
+			req.Header.Set("Authorization", "Bearer 123")
+
+			handlerFunc(rec, req)
+
+			Expect(rec.Code).To(Equal(http.StatusUnauthorized))
+		})
+
+		It("should return 401 Unauthorized when token is invalid", func() {
```
Comment (piksel) [inline pkg/api/api_test.go]:
```suggestion
		It("should return 200 OK when token is valid", func() {
```

**Review by piksel — APPROVED** [review]
Looks really good. I have no idea how the commit manages to "decreases coverage by 0.46%" while also having 100% diff coverage...

File: pkg/api/api_test.go (hunk 1 above)
Comment (hypnoglow) [reply pkg/api/api_test.go]: Nice catch, thanks. Fixed.

**Review by piksel — APPROVED** [review]

**Comment by piksel** [comment]
@all-contributors please add @hypnoglow for code

**Pull request merged**